src/com/yashu/interview/map/linkedhashset/HashSet.java

```diff
@@ -0,0 +1,218 @@
+package HashSet;
+import java.util.InputMismatchException;
+public class HashSet<T> implements OrderedSet<T>, Container<T>{
+
+	private	Node<T>[] array;
+	private ArrayIndexCalculator indexCalculator;
+	private LinkedList<T> linkedList;
+	private int count = 0;
+
+
+	/**
+	 * Parameterized constructor
+	 * @param initialSize - Initial size of the set
+	 */
+	@SuppressWarnings("unchecked")
+	public HashSet(int initialSize){
+		array = new Node[initialSize];
+		indexCalculator = new ArrayIndexCalculator();
+		linkedList = new LinkedList<T>();
+	}
+	 
+	// Package visible only - for testing purposes
+	@SuppressWarnings("unchecked")
+	HashSet(int initialSize, ArrayIndexCalculator indexCalculator) {
+		array = new Node[initialSize];
+		this.indexCalculator = indexCalculator;
+		linkedList = new LinkedList<T>();
+	}
+	// Returns the size of the Array.
+	public int getLengthOfArray() {
+		return array.length;
+	}
```

Comment:
Remove getLengthOfArray() method. It doesn't need to be public and you get the length by simply calling array.length wherever you need.